kickoff/jsonexportl10n.py

```diff
@@ -74,12 +101,33 @@ def jsonRegionsExports():
 
 def generateListPerProduct(product):
     releases = getReleases(ready=True, productFilter=product)
-    version_list = []
+    registrar = _L10nReleasesRegistrar()
     for r in releases:
-        if r._shippedAt and r.l10nChangesets != config.LEGACY_KEYWORD:
-            # Only list version which shipped with l10n content
-            version_list.append(r.name)
-    return version_list
+        registrar.addRelease(r)
+    return registrar.releases
+
+
+class _L10nReleasesRegistrar:
+    # Matches strings like "Firefox-19.0b1-build1" and stores "Firefox-19.0b"
+    BETA_REGEX = re.compile(r"([^\-]+-(\d+\.)+0b)\d+-build\d+")
+
+    def __init__(self):
+        self.releases = []
+        self._betas_already_processed = set()
+
+    def addRelease(self, release):
+        if release.isShippedWithL10n:
+            self._addAggregatedBetaOnlyOnce(release)
+            self.releases.append(release.name)
+
+    def _addAggregatedBetaOnlyOnce(self, release):
+        beta_name_match = self.BETA_REGEX.match(release.name)
+        if beta_name_match is not None:
```

Comment:
I don't mind changing to something shorter.

Nonetheless, `re.match()` [explicitly returns None](https://docs.python.org/2/library/re.html#re.match) when no pattern was matched. [Per PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations), we should compare against None with the `is` keyword.

I don't know if the internals of `==` has been improved since 2009, but `is` [has been experienced to be faster](http://jaredgrubb.blogspot.fr/2009/04/python-is-none-vs-none.html) than `==`.